### DISCUSSION TITLE
How to get request's path with parameter not substituted?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [pallets/flask#5454] [Q&A] [answered]
Hello,
I hope your day is going well.
I'm working with Flask endpoints that have parameters, e.g. `@app.route("/prefix/<param>", methods=["GET"])`. Suppose param can be a string. I'm looking for a way to access a current route's parametrized path, i.e. `/prefix/<param>`, not `/prefix/abc`. How can I do that? What would `request.path` return?
Thank you.

**Comment by davidism** [comment, marked as answer]
`request.url_rul` contains the matched `Rule` object https://werkzeug.palletsprojects.com/en/3.0.x/routing/#werkzeug.routing.Rule

**Reply by andrii-korotkov-verkada** [reply]
Thank you! `url_rule` is the answer.
